pkg/types/gateway/trigger.go

```diff
@@ -32,3 +37,107 @@ type HTTPTriggerResponse struct {
 	WorkflowExecutionID string            `json:"workflow_execution_id,omitempty"` // ID of the workflow execution.
 	Status              HTTPTriggerStatus `json:"status,omitempty"`                // Status of the trigger request.
 }
+
+// MarshalJSON implements custom JSON marshalling to ensure alphabetical order of keys for WorkflowSelector,
+// and only includes non-empty fields.
+func (ws WorkflowSelector) MarshalJSON() ([]byte, error) {
+	m := make(map[string]interface{})
+	if ws.WorkflowID != "" {
+		m["workflowID"] = ws.WorkflowID
+	}
+	if ws.WorkflowName != "" {
+		m["workflowName"] = ws.WorkflowName
+	}
+	if ws.WorkflowOwner != "" {
+		m["workflowOwner"] = ws.WorkflowOwner
+	}
+	if ws.WorkflowTag != "" {
+		m["workflowTag"] = ws.WorkflowTag
+	}
+	return marshalWithSortedKeys(m)
+}
+
+// MarshalJSON implements custom JSON marshalling to ensure deterministic output
+// with sorted keys at all levels for map[string]interface{}, including nested objects in the Input field.
+func (r HTTPTriggerRequest) MarshalJSON() ([]byte, error) {
+	result := make(map[string]interface{})
```

Comment:
why not use `any`?

Reply:
no strong preference here. can change to any